### PR TITLE
Endpoints and types refactor for solana

### PR DESCRIPTION
**Pull request by b4rtaz**
---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [ ] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [ ] I have made corresponding changes to the documentation
- [ ] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: packages/core/src/operations/Operation.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { MoralisCore } from '../MoralisCore';
+
+export interface Operation<Request, JSONRequest, Response, JSONResponse> {
+  name: string;
+  groupName: string;
+  method: OperationRequestMethod;
+  urlPathPattern: string;
+  urlPathParamNames: (keyof Request)[];
+  urlSearchParamNames?: (keyof Request)[];
```
Comment (Y0moo) [inline packages/core/src/operations/Operation.ts:9]: I see that SolApi has only `path` params and no `search` from open API types, so the `Request` generic type here consists only of `path`. But in the future EvmApi refactoring don't you think it'd be more accurate to split `Request` to `PathParams` and `SearchParams` instead?

Comment (b4rtaz) [reply packages/core/src/operations/Operation.ts:9]: Check [getWalletTokenTransfersOperation.ts](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/endpoints-and-types-refactor-for-solana/packages/common/evmUtils/src/operations/token/getWalletTokenTransfersOperation.ts) and [runContractFunctionOperation.ts](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/endpoints-and-types-refactor-for-solana/packages/common/evmUtils/src/operations/utils/runContractFunctionOperation.ts) files. There are search parameters and body parameters.

File: packages/core/src/operations/OperationRequestValidator.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  public validate(request: Request) {
+    const requestParamNames = Object.keys(request as Record<string, unknown>);
+
+    for (const paramName of requestParamNames) {
+      if (!this.allParamNames.includes(paramName as keyof Request)) {
+        throw new MoralisError({
+          code: CoreErrorCode.INVALID_ARGUMENT,
+          message: `Request contains unknown parameter: ${paramName}. This operation supports the fallowing parameters: ${this.allParamNames.join(
```
Comment (ErnoW) [inline packages/core/src/operations/OperationRequestValidator.ts:22]:
```suggestion
          message: `Request contains unknown parameter: ${paramName}. This operation supports the following parameters: ${this.allParamNames.join(
```

**Review by ErnoW — COMMENTED** [review]
Looks very good 👏 . Only some small nitpicks.

**Review by ErnoW — APPROVED** [review]

**Pull request merged**